Presentation.Web/app/components/it-system/edit/tabs/it-system-tab-reference-create-modal.controller.ts

```diff
@@ -1,52 +1,86 @@
-﻿((ng, app) => {
-    app.config(["$stateProvider", $stateProvider => {
-        $stateProvider.state("it-system.edit.references.create", {
-            url: "/createReference/:id",
-            onEnter: ["$state", "$uibModal", ($state, $modal) => {
-                $modal.open({
-                    templateUrl: "app/components/it-reference/it-reference-modal.view.html",
-                    // fade in instead of slide from top, fixes strange cursor placement in IE
-                    // http://stackoverflow.com/questions/25764824/strange-cursor-placement-in-modal-when-using-autofocus-in-internet-explorer
-                    windowClass: "modal fade in",
-                    controller: "system.referenceCreateModalCtrl",
-                    resolve: {
-                        referenceService: ["referenceServiceFactory", (referenceServiceFactory) => referenceServiceFactory.createSystemReference()],
-                    }
+﻿module Kitos.ItSystem.Catalog.Edit.Reference.Create {
+    "use strict";
+
+    export class CreateReferenceItSystemEditController {
+        static $inject: Array<string> = [
+            "$scope",
+            "notify",
+            "$state",
+            "$stateParams",
+            "$uibModalInstance",
+            "referenceService"
+
+        ];
+
+        constructor(
+            private readonly $scope,
+            private readonly notify,
+            private readonly $state: angular.ui.IStateService,
+            private readonly $stateParams,
+            private readonly $uibModalInstance: ng.ui.bootstrap.IModalServiceInstance,
+            private referenceService: Services.ReferenceService) {
+
+            $scope.modalTitle = "Opret reference";
+        }
+
+        save() {
+
+            var msg = this.notify.addInfoMessage("Gemmer række", false);
 
-                }).result.then(() => {
-                    // OK
-                    // GOTO parent state and reload
-                    $state.go("^", null, { reload: true });
-                }, () => {
-                    // Cancel
-                    // GOTO parent state
-                    $state.go("^");
-                });
+            this.referenceService.createReference(
+                this.$stateParams.id,
+                this.$scope.reference.title,
+                this.$scope.reference.externalReferenceId,
```

Comment:
koden i denne og de andre varianter er jo præcis den samme.. det er kun din resolve (create System|Contract|Project)Service der varierer.... og det gælder også for din "edit modal" dvs. du kan nu lave en create + edit controller og smide ned ved siden af det generiske view og i din state definition kan du så pege på den controller i stedet for at copy paster alt det her kode. Så genbruger dine filer rent faktisk noget